### PR TITLE
Create Initial Web Application Files 

### LINKED ISSUE #2: Create Barebones Structure for Repository
Create the maven structure with the barebones Spring and React files embedded.

Priority: P0
Requirement Reference: 2

### LINKED ISSUE #18: Setup React Project
Setup Project Structure for React Project

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR integrates the minimal files needed to run the React web application.

- _components_ is the folder designated to hold any supporting React Components to be used within the main _App_ component
- _index.js_ is the main JavaScript file used to render our _App_ component to the DOM
- _package.json_ indicates node dependencies and some supporting information about the project.
**Please review the supporting information in package.json such as name, description, author and license**
- _index.html_ is our entry point
- _style.css_ the main stylesheet for index.html

Closes #18 
Closes #2

### CODE REVIEW

File: webapp/App.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as React from "react";
+
+/**
+ * Parent Component that is Rendered to the Page.
+ * @param {*} props arbituary input properties for our application to use
+ */
+export const App = (props) => {
+    return (
```
Comment (MagoDelBlocco) [inline webapp/App.js:8]: consider using [tab sizes of 2 spaces](https://google.github.io/styleguide/jsguide.html#formatting-block-indentation)

File: webapp/App.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+        <h1>Welcome {props.name}</h1>
+    )
+}
```
Comment (MagoDelBlocco) [inline webapp/App.js:11]: Please add a [newline at the end of the file](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file)

**Comment by Somae** [comment]
This pull request has no approving review from a pod member yet. Please only add Krasimir or I once that is the case, unless you want to discuss something on the pull request, then tag us why you added us in advance. Thank you!

**Comment by holounic** [comment]
Consider adding newline at the end of files

File: webapp/App.js (hunk 1 above)
Comment (samuelsogbesan) [reply webapp/App.js]: 207b340

File: webapp/App.js (hunk 2 above)
Comment (samuelsogbesan) [reply webapp/App.js]: commit 927ef05

File: webapp/App.js (hunk 1 above)
Comment (MagoDelBlocco) [reply webapp/App.js]: ditto

I believe you are using two [tabs](http://tarantsov.com/hackers-coding-style-guide/why-tabs-should-be-avoided.html#:~:text=This%20actually%20requires%20mixing%20tabs,like%20in%20the%20example%20above.) right now, not two spaces.

**Comment by samuelsogbesan** [comment]
@MagoDelBlocco Made some more changes, all files should use 2 space now.

File: src/main/webapp/App.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import * as React from "react";
+
+/**
+ * Parent Component that is Rendered to the Page.
```
Comment (krasimir-google) [inline src/main/webapp/App.js:4]: nit: [per javadoc conventions](https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs) (I realize this is js-doc, but close enough), add an empty line (i.e. one that only contains `*`) between the summary section and the `@param` section.

Comment (samuelsogbesan) [reply src/main/webapp/App.js:4]: Updated with commit 793c906

File: src/main/webapp/index.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<html>
+    <head>
+        <link rel="stylesheet" href="style.css">
+    </head>
+
+    <body>
+        <div id="root"></div>
+        <script defer src="./index.js"></script>
+    </body>
+</html>
```
Comment (Somae) [inline src/main/webapp/index.html:10]:
```suggestion
</html>

```

File: src/main/webapp/index.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as React from "react";
+import {render} from "react-dom";
+import {App} from "./App";
+
+// Render our App Component to element with ID == target
+const target = 'root';
+render (
+  <App name = 'Sam'/>,
```
Comment (Somae) [inline src/main/webapp/index.js:8]: Did you name this app Sam for testing purposes?

File: src/main/webapp/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  "name": "llvm-dashboard",
+  "version": "1.0.0",
+  "description": "a dashboard for viewing the build status of buildbots for the llvm project",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "dev": "parcel index.html",
+    "build": "parcel build index.html"
+  },
+  "author": "",
```
Comment (Somae) [inline src/main/webapp/package.json:11]: All Graph Geckos are authors of this project.

**Review by Somae — CHANGES_REQUESTED** [review]

File: src/main/webapp/index.js (hunk 5 above)
Comment (samuelsogbesan) [reply src/main/webapp/index.js:8]: Perhaps the name of the prop is ambigious. "name" here is for testing purposes, and refers to a name of a person that is later rendered to screen as the string "Welcome [name]". See _App.js_ also.

File: src/main/webapp/package.json (hunk 6 above)
Comment (samuelsogbesan) [reply src/main/webapp/package.json:11]: Should I list that as Graph Geckos, or as our names respectivelly?

Comment (samuelsogbesan) [reply src/main/webapp/package.json]: I had a look around, and it seems that package.json's author field is only supposed to be used for a single author. Do you suggest I write my own name there, or "Graph Geckos"?

Comment (samuelsogbesan) [reply src/main/webapp/package.json]: Found a field called contributors - I'll use that instead.

File: webapp/App.js (hunk 2 above)
Comment (samuelsogbesan) [reply webapp/App.js]: Updated with commit 21cd082

File: webapp/App.js (hunk 1 above)
Comment (samuelsogbesan) [reply webapp/App.js]: Updated with commit 927ef05

**Review by Somae — APPROVED** [review]

**Review by MagoDelBlocco — APPROVED** [review]

**Pull request merged**